tests/Sub/SqsSnsQueueTest.php

```diff
@@ -70,124 +59,154 @@ protected function setUp(): void
     /** @test */
     public function it_can_instantiate_the_queue()
     {
-        $queue = new SqsSnsQueue($this->mockedSqsClient, 'default');
+        $queue = new SqsSnsQueue(m::mock(SqsClient::class), 'default');
 
         $this->assertInstanceOf(SqsSnsQueue::class, $queue);
     }
 
     /** @test */
-    public function it_can_receive_a_message_and_pop_it_off_the_queue()
+    public function it_can_receive_a_message_and_pop_it_off_the_queue_which_should_dispatch_an_internal_event()
     {
-        $this->mockedSqsClient->method('receiveMessage')->willReturn(new \Aws\Result([
+        Event::fake();
+        $sqs = m::mock(SqsClient::class);
+        $sqs->shouldReceive('receiveMessage')->once()->andReturn(new \Aws\Result([
             'Messages' => [$this->mockedMessageDataWithTopicOnly],
         ]));
 
-        $queue = new SqsSnsQueue($this->mockedSqsClient, 'default', '', '', [
-            'TopicArn:123456' => '\\TopicListener',
-        ]);
-        $queue->setContainer(new Container);
-        $job = $queue->pop();
+        $queue = new SqsSnsQueue($sqs, 'default');
+        $queue->setContainer($this->app);
+        $queue->pop();
 
-        $this->assertInstanceOf(SqsSnsJob::class, $job);
+        Event::assertDispatched('TopicArn:123456');
```

Comment:
It's not immediately obvious where this is coming from (same with the subject and payload in the tests below)

Perhaps we can replace `mockedMessageDataWithTopicOnly` with a factory that can allow us to specify the data we care about in the test